lib/castle/core/process_response.rb

```diff
@@ -36,8 +37,25 @@ def verify!(response)
 
           raise Castle::InternalServerError if response.code.to_i.between?(500, 599)
 
+          raise error422(response) if response.code.to_i == 422
+
           error = RESPONSE_ERRORS.fetch(response.code.to_i, Castle::ApiError)
-          raise error, response[:message]
+
+          raise error
+        end
+
+        def error422(response)
+          if response.body
+            begin
+              parsed_body = JSON.parse(response.body, symbolize_names: true)
+              if parsed_body.is_a?(Hash) && parsed_body[:type] == INVALID_REQUEST_TOKEN
+                return Castle::InvalidRequestTokenError
+              end
+            rescue JSON::ParserError
```

Comment:
Do we want to assume our API can return an invalid JSON here? Perhaps we could catch `JSON::ParserError`  in the `call` method instead.

Reply:
this is also used for the legacy endpoints, I wanted to keep the functionality unchanged if that error type value is not present

Reply:
Legacy APIs respond with the same format for 422 errors

Reply:
not sure about eg impersonation endpoints, I prefer to keep that somehow covered